sympy/polys/polytools.py

```diff
@@ -3773,6 +3774,178 @@ def nth_power_roots_poly(f, n):
+            nonzero roots.
+
+        Notes
+        =====
+
+        The domain of the polynomial must be a subring of the complex numbers.
+        We use the `Cauchy bound`_.
+
+        .. _Cauchy bound: https://en.wikipedia.org/wiki/Geometrical_properties_of_polynomial_roots#Lagrange's_and_Cauchy's_bounds
+
+        """
+        dom = f.get_domain()
+        if dom.is_FiniteField or not dom.is_Numerical:
+            raise DomainError("Cannot compute root bounds over %s" % dom)
+        if f.is_multivariate:
+            raise MultivariatePolynomialError(
+                "Must be a univariate polynomial")
+        if f.is_monomial:
+            raise PolynomialError("Monomial has no nonzero roots")
+
+        coeffs = f.all_coeffs()
+        while coeffs[-1] == 0:
+            coeffs.pop()
+        lc, tc = coeffs[0], coeffs[-1]
+        ub = 1 + max(abs(n / lc) for n in coeffs[1:])
+        lb_inv = 1 + max(abs(n / tc) for n in coeffs[:-1])
+
+        return 1/lb_inv, ub
+
+    def root_separation_lower_bound(f):
+        """
+        Compute a lower bound on the minimum distance between any two
+        roots of this polynomial.
+
+        Examples
+        ========
+
+        >>> from sympy import Poly
+        >>> from sympy.abc import x
+        >>> Poly(x**2 - 2).root_separation_lower_bound()
+        1
+        >>> (x * Poly(x**2 - 2)).root_separation_lower_bound()
+        12/17
+
+        Returns
+        =======
+
+        :py:class:`~.Expr`
+            A non-negative lower bound which will be positive if all the roots
+            are distinct, and rational if the coefficients of the polynomial
+            are rational.
+
+        Raises
+        ======
+
+        DomainError
+            If the domain of the polynomial is not a subring of the complex
+            numbers.
+        MultivariatePolynomialError
+            If the polynomial is not univariate.
+
+        """
+        dom = f.get_domain()
+        if dom.is_FiniteField or not dom.is_Numerical:
+            raise DomainError("Cannot compute root bounds over %s" % dom)
+        if f.is_multivariate:
+            raise MultivariatePolynomialError(
+                "Must be a univariate polynomial")
+        if f.is_monomial:
+            return Integer(0)
+
+        x = f.gen
+        y = Dummy('y')
+        g = f.subs(x, x + y)
+        h = Poly(resultant(f, g, gens=[x, y]), domain=f.domain)
+        # If f has roots ri, the roots of h are all the ri - rj.
+
+        if h.is_monomial:
+            # In this case all roots of f are the same.
+            return Integer(0)
+
+        lb, _ = h.root_bounds()
+
+        # Since h is an even function, can improve bound with square root.
+        if lb.is_rational:
+            # Preserve rational bound
+            return Rational(isqrt(lb.p*lb.q) + 1, lb.q)
+        else:
+            from sympy.functions.elementary.miscellaneous import sqrt
+            return sqrt(lb)
```

Comment:
If the function just returns the square of the bound then the shenanigans here are not needed and the return value will always be within the field generated by the coefficients.

Reply:
In other words, you're saying that sure the square root sharpens the bound, but who cares. Maybe it's not worth the extra calculation. I suppose that may be true in many cases.

Maybe the method should take a keyword arg, `take_sqrt=False` so those who care can activate it.

Reply:
The sharp bound is good. It's just that I can have a function that returns the bound or a function that returns the square of the bound and each is equally usable with the same sharpness:
```python
if abs(value) > bound():
    # do stuff

if value ** 2 > bound_squared():
    # do the same stuff
```
The difference between these two cases is that in the latter I can do all calculations within the same field e.g. if `bound_squared()` is guaranteed to return a rational number and value is a rational number then the check `value ** 2 > bound_squared()` is pure rational arithmetic and does not require computing in an extension field or using floating point.

In the complex case even `abs(value)` involves a square root but `abs(value)**2` is `re(value)**2 + im(value)**2` so if both parts are rational then the result is rational.

Reply:
Okay, those are some excellent points. I suppose as long as the method is called something like `root_separation_lower_bound_squared()`, users will know what they're getting. Also the examples in the method's docstring can show the usage patterns you just described.